### PR TITLE
[CodeClean] fix svace issue

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Code clean, fix svace issues about lock in single/pipeline handle.

### CODE REVIEW

**Review by again4you — APPROVED** [review]
LGTM!

File: c/src/ml-api-inference-pipeline.c [hunk 1]
```diff
@@ -1777,14 +1777,21 @@ get_app_src_callback (ml_pipeline_common_elem * src_h, void **data)
 {
   ml_pipeline_src_callbacks_s *src_cb = NULL;
   ml_pipeline_element *elem;
+  ml_pipeline *p;
 
+  p = src_h->pipe;
   elem = src_h->element;
+
+  g_mutex_lock (&p->lock);
```
Comment (myungjoo) [inline c/src/ml-api-inference-pipeline.c:1785]: What's the op/data protected by this lock in line 1788-1791?

Comment (jaeyun-jung) [reply c/src/ml-api-inference-pipeline.c:1785]: This function returns appsrc callback which sets with pipeline lock, so svace-checker returns error when getting cb info without p-lock.

**Review by niley7464 — APPROVED** [review]
LGTM!

File: c/src/ml-api-inference-pipeline.c (hunk 1 above)
Comment (myungjoo) [reply c/src/ml-api-inference-pipeline.c:1785]: The problem is that... ```ml_pipeline_common_elem``` has ```{ callback_info_s , ml_pipeline }```
and ```ml_pipeline->lock``` is protecting ```callback_info_s->*```.

this lock should protect elements in ml_pipeline, not others in ml_pipeline_common_elem.
If it really needs to protect  callback_info_s, it means the lock is really messed up.

Comment (myungjoo) [reply c/src/ml-api-inference-pipeline.c:1785]: The lock needs to declare (in doxygen) which are protected by the lock.

**Review by songgot — APPROVED** [review]
LGTM

**Review by myungjoo — APPROVED** [review]

**Pull request merged**